dask/bag/core.py

```diff
@@ -1630,8 +1648,14 @@ def from_delayed(values):
     return Bag(graph, name, len(values))
 
 
-def merge_distinct(seqs):
-    return set().union(*seqs)
+@curry
+def merge_distinct(seqs, key=None):
+    if key is None:
+        return list(toolz.unique(toolz.concat(seqs)))
```

Comment:
If cytoolz is around then we'll probably want to use cytoolz.unique instead.  See the toolz and cytoolz imports at the top of this file.